mypy/messages.py

```diff
@@ -569,13 +577,13 @@ def no_variant_matches_arguments(self,
         num_args = len(arg_types)
         if num_args == 0:
             self.fail('All overload variants{} require at least one argument'.format(name_str),
-                      context)
+                      context, code=codes.CALL_OVERLOAD)
         elif num_args == 1:
             self.fail('No overload variant{} matches argument type {}'
-                      .format(name_str, arg_types_str), context)
+                      .format(name_str, arg_types_str), context, code=codes.CALL_OVERLOAD)
         else:
             self.fail('No overload variant{} matches argument types {}'
-                      .format(name_str, arg_types_str), context)
+                      .format(name_str, arg_types_str), context, code=codes.CALL_OVERLOAD)
```

Comment:
This would need some magic or global state and I'd rather avoid both. It can also be error-prone in case we call other functions that can call `fail`. I think that explicit is better than implicit here.

Reply:
It is debatable what is more error-prone: having some state for error reporting (we actually already have some in e.g. `SemanticAnalyzer.file_context()` etc) or having to repeat codes many times. This is however not important to delay this, we can easily add this later if someone likes the idea.